integration_tests/src/main/python/window_function_test.py

```diff
@@ -1066,3 +1066,24 @@ def test_unbounded_to_unbounded_window():
     assert_gpu_and_cpu_are_equal_collect(lambda spark : spark.range(1024).selectExpr(
         'SUM(id) OVER ()',
         'COUNT(1) OVER ()'))
+
+_nested_gens = array_gens_sample + struct_gens_sample + map_gens_sample
```

Comment:
Can we add some tests for nth too. Not just first and last? And with this can we test negative input offsets? From the code it looks like offsets must be >= 0. This would be here just to catch cases in the future if Spark expands into negative offsets.

Reply:
> ... can we test negative input offsets? From the code it looks like offsets must be >= 0.

I suspect it is unlikely that Spark would support negative offsets for `NTH_VALUE()`. The convention in Oracle, Hive, Snowflake, etc. seems to be that `N` be positive.

Reply:
Then as a nit I would like an assertion in the code to make sure that it follows the given guides too. This is just being defensive.